internal/wasm/jit/engine_test.go

```diff
@@ -477,3 +471,114 @@ func TestJIT_SliceAllocatedOnHeap(t *testing.T) {
 		})
 	}
 }
+
+func TestEngine_CachedCompiledFunctionsPerModule(t *testing.T) {
```

Comment:
```suggestion
// TODO: move most of this logic to enginetest.go so that there is less drift between interpreter and jit
func TestEngine_CachedCompiledFunctionsPerModule(t *testing.T) {
```